lib/index.js

```diff
@@ -0,0 +1,18 @@
+module.exports = throttlingPlugin
+
+const Bottleneck = require('bottleneck/light')
+const wrapRequest = require('./wrap-request')
+
+function throttlingPlugin (octokit) {
+  const globalLimiter = new Bottleneck({
+    maxConcurrent: 1
+  })
+  const writeLimiter = new Bottleneck({
+    minTime: 1000
+  })
+
+  octokit.throttle = {}
+  const emitter = new Bottleneck.Events(octokit.throttle)
+
+  octokit.hook.wrap('request', wrapRequest.bind(null, emitter, globalLimiter, writeLimiter))
```

Comment:
I usually create a single `state` variable with properties that I pre-bind as the first argument, so in that case I’d do 

```suggestion
  const state = {
    emitter,
    globalLimiter,
    writeLimiter
  }
  octokit.hook.wrap('request', wrapRequest.bind(null, state))
```

That way I don’t have to remember the order and number of pre-bound variables 🤪

Reply:
It's already just `emitter` and `state` in a later commit, but I'll put `emitter` into `state` too.